numpy/core/meson.build

```diff
@@ -691,15 +684,31 @@ py.extension_module('_multiarray_tests',
   subdir: 'numpy/core',
 )
 
+mtargets_res = mod_features.multi_target(
+  '_umath_tests.dispatch.h',
+  'src/umath/_umath_tests.dispatch.c',
+  dispatch: [
+    AVX2, SSE41, SSE2,
+    ASIMDHP, ASIMD, NEON,
+    VSX3, VSX2, VSX,
+    VXE, VX,
```

Comment:
Just checking: the ordering of the above 4 lines is just for clarity right? There's no longer any parsing of this, it's just a list of features across all architectures that should be used if the hardware and build options support it?

Reply:
> Just checking: the ordering of the above 4 lines is just for clarity right? 

Indeed, they are separated for clarity only but the order of CPU features that related to each architecture is matters.
For example, if `SEE41` is specified before `AVX2` then AVX2 will be ignored. The reason behind this behavior is the module provide the ability to concat multiple features that don't imply each other for single target. e.g. `[FMA3, AVX2]` or `[ASIMDHP, ASIMDDP]`  

> There's no longer any parsing of this, it's just a list of features across all architectures that should be used if the hardware and build options support it?

Yes its true, `multi_targets()` test each specified feature before enable it. If the feature is disabled or not supported it will be escaped and provides the reason behind escaping these features through the debug log.

The main `meson_cpu/meson.build` disables any features that are not supported by platform(to avoid unnecessary tests) or the disabled by build options via:
https://github.com/numpy/numpy/blob/505e649200d54bdc653cc11927b6cac4dc1729b3/meson_cpu/meson.build#L240-L248

Reply:
Thanks, sounds good!

> For example, if `SEE41` is specified before `AVX2` then `AVX2` will be ignored.

Should that reverse ordering simply raise an error in the configure stage then? If `AVX2` would always be silently ignored if it's behind `SSE41`, then it's either user error or `AVX2` shouldn't be there at all.

Reply:
> For example, if SEE41 is specified before AVX2 then AVX2 will be ignored.

Sorry, that wasn't accurate what going to happen is the following, an object will be generated for `AVX2` but the runtime dispatching call will gives the priority for `SSE41` over `AVX2`.

> Should that reverse ordering simply raise an error in the configure stage then? 

I have updated the module to validate this behavior and provides an argument to bybass this validation, see:

https://github.com/mesonbuild/meson/blob/3e634e5bbdeae8d32330b585c98898aa0cfb11fa/mesonbuild/modules/features/module.py#L470-L479

We can force autosort these targets based on highest interest similar to the disutils but as I said before it isn't accurate 100% when it comes to multi-features per target.

And the reason why is simply because of this function:
https://github.com/mesonbuild/meson/blob/3e634e5bbdeae8d32330b585c98898aa0cfb11fa/mesonbuild/modules/features/module.py#L611-L631
or 
https://github.com/numpy/numpy/blob/09ee7daa9113ecb503552913d76185d0ad3f0f31/numpy/distutils/ccompiler_opt.py#L1336-L1345